webapp/channels/src/selectors/rhs.ts

```diff
@@ -138,6 +138,37 @@ export function getIsSearchGettingMore(state: GlobalState): boolean {
     return state.entities.search.isSearchGettingMore;
 }
 
+export function makeGetDraft() {
+    let defaultDraft = {message: '', fileInfos: [], uploadsInProgress: [], createAt: 0, updateAt: 0, channelId: '', rootId: ''};
+    return (state: GlobalState, channelId: string, rootId = ''): PostDraft => {
+        if (defaultDraft.channelId !== channelId || defaultDraft.rootId !== rootId) {
+            defaultDraft = {message: '', fileInfos: [], uploadsInProgress: [], createAt: 0, updateAt: 0, channelId, rootId};
```

Comment:
Something about changing a default value here seems really weird to me. I get that you're trying to reuse the same default object between calls, but this looks odd to me.

Perhaps this could be rewritten using `createSelector` to make it follow a more standard pattern. We wouldn't necessarily storing the `defaultDraft` between calls to the selector, but I think we could write it in a way that we still wouldn't return a new object unless the stored draft changes

Reply:
I will take a look, but I think it would be nice to discuss this in sync, to make sure we are aligned with the pattern we want to use :smile: